### ISSUE TITLE
Drift for warehouse size 2XLarge

### ISSUE THREAD

**Issue by mwufigma** [Snowflake-Labs/terraform-provider-snowflake#2348]
### Terraform CLI and Provider Versions

Snowflake provider version 0.68.0
Terraform version 1.5.0

### Terraform Configuration

```terraform
resource "snowflake_warehouse" "wh" {
  name              = "elt_wh"
  comment           = ""
  warehouse_size    = "2XLARGE"
  max_cluster_count = 2
  auto_suspend      = 60
}
```

### Expected Behavior

Apply should go through successfully and warehouse should be updated to 2XLARGE.

### Actual Behavior

If I use `"2XLARGE"` as the `warehouse_size` value, I get this error:
```
│ Error: not a valid warehouse size: 2XLARGE
│ 
│   with snowflake_warehouse.creation_dataset_elt,
│   on warehouses.tf line 101, in resource "snowflake_warehouse" "creation_dataset_elt":
│  101:   warehouse_size    = "2XLARGE"
```

However, if I use the accepted value `"xxlarge"`, Terraform produces drift like this:
```
Terraform will perform the following actions:

  # snowflake_warehouse.creation_dataset_elt will be updated in-place
  ~ resource "snowflake_warehouse" "wh" {
        id                                  = "elt_wh"
        name                                = "elt_wh"
      ~ warehouse_size                      = "2XLARGE" -> "xxlarge"
        # (12 unchanged attributes hidden)
    }
```

### Steps to Reproduce

1. Create an 2XL warehouse through terraform using any of the values I supplied above (`"2XLARGE"` or `"xxlarge"`)

### How much impact is this issue causing?

High

### Additional Information

This is causing perma-drift in our repo and messes up Spacelift (our Terraform module management software)

**Comment by sfc-gh-asawicki** [comment]
Hey @mwufigma. Thanks for reporting the issue.

It was already solved with https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/1946. It's available since v0.68.2. Please bump and confirm.

**Comment by sfc-gh-asawicki** [comment]
Hey @mwufigma, I will close this issue; please open a new one if you are experiencing this problem in the more recent versions of the provider.
